handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -494,6 +494,12 @@ public void flush(ChannelHandlerContext ctx) throws Exception {
         }
         try {
             wrap(ctx, false);
+        } catch (Exception e) {
+            // Just fail the pending writes in case of an SSLException. If there are no pending writes rethrow.
+            if (pendingUnencryptedWrites.removeAndFailAll(e)) {
```

Comment:
> I am still not sure I understand what "use-case" you are talking about...

Do you have a practical example of when we expect to hit line 500?

I agree we should not swallow exceptions and that we should fail the write promises if we cannot complete the writes. I'm just wondering why we don't always throw the exception here and if we really need the changes to `PendingWriteQueue`:

```java
pendingUnencryptedWrites.removeAndFailAll(e);
throw e;
```